src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -1818,9 +1820,16 @@ public SocketError SendToAsync(Memory<byte> buffer, int offset, int count, Socke
 
             bytesSent = 0;
             SocketError errorCode;
-            int observedSequenceNumber;
-            if (_sendQueue.IsReady(this, out observedSequenceNumber) &&
-                SocketPal.TryCompleteSendTo(_socket, buffer.Span, ref offset, ref count, flags, socketAddress, socketAddressLen, ref bytesSent, out errorCode))
+            int observedSequenceNumber = 0;
```

Comment:
StartAsyncOperation needs to try complete the operation when the speculative attempt fails. You need to make some more changes, because it doesn't happen when `_sequenceNumber` == `observedSequenceNumber` == `0`.